avocado/core/job.py

```diff
@@ -127,10 +145,16 @@ def __init__(self, config=None):
                        line options and argument parsing
         :type config: dict
         """
+        core_config = settings.as_dict()
+        if config:
+            config.update(core_config)
+            self.config = config
+        else:
+            self.config = core_config
```

Comment:
I might be wrong, be maybe here we have some problems:

The default settings (`settings.as_dict()`) are overriding the "user's config intention", right?

```python3
core_config = {'a': 1} # Default value for a
config = {'a': 2} # User would like to use a = 2
config.update(core_config)
config
{'a': 1}
```

Maybe IIUC, the idea is something like that:


```python
self.config = settings.as_dict()
if config:
   # yes, it is strange but with python 3.9 we will be able to use `|` operator for dicts
   self.config = {**self.config, **config}
```

or if the `**` is too ugly:

```python
self.config = settings.as_dict()
if config:
   merged_config = self.config.update(config)
   self.config = merged_config
```
Makes sense?

Reply:
True, I got the update on the wrong order (not the first time). Thanks for catching that.

Reply:
FIY @beraldoleal , there's a small logic issue with your proposal too, because `update()` doesn't return anything, it operates in place.  So this seems to be enough:

```python
        self.config = settings.as_dict()
        if config:
            self.config.update(config)
```

Reply:
Even better.